osu.Game/IO/MigratableStorage.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using osu.Framework.Platform;
+
+namespace osu.Game.IO
+{
+    /// <summary>
+    /// A <see cref="WrappedStorage"/> that is migratable to different locations.
+    /// </summary>
+    public abstract class MigratableStorage : WrappedStorage
+    {
+        internal virtual string[] IgnoreDirectories => Array.Empty<string>();
+        internal virtual string[] IgnoreFiles => Array.Empty<string>();
```

Comment:
This is a weird modifier mix. Any consumers from a different assembly that might want to override this will not be able to. I think this calls for the [rarely-seen](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/protected-internal)

```suggestion
        protected internal virtual string[] IgnoreDirectories => Array.Empty<string>();
        protected internal virtual string[] IgnoreFiles => Array.Empty<string>();
```

(Or heck, just make it public, I don't think that's that much of a deal.)

Needs to be fixed up in the override too if this happens.

Reply:
I've changed it to public for now, will await further feedback about this in case `protected` becomes the preferred way to go.